src/stores/AppStore/AppStore.ts

```diff
@@ -2108,6 +2115,38 @@ export class AppStore {
         autorun(() => {
             this.activateStatsPanel(this.preferenceStore.statsPanelEnabled);
         });
+
+        // listen devicePixelRatio
+        let remove = null;
+        const updatePixelRatio = () => {
+            if (remove != null) {
+                remove();
+            }
+            const mqString = `(resolution: ${window.devicePixelRatio}dppx)`;
+            const media = matchMedia(mqString);
+            media.addEventListener("change", updatePixelRatio);
+            remove = () => {
+                media.removeEventListener("change", updatePixelRatio);
+            };
+
+            this.handleDevicePixelRatioChange(this.devicePixelRatio);
+        };
+        updatePixelRatio();
+    }
+
+    // update devicePixelRatio and make the image size invariant on screen
+    @action private handleDevicePixelRatioChange(prevPixelRatio: number) {
```

Comment:
Perhaps we should rename `prevPixelRatio` to `prevDevicePixelRatio` to be clear that it is not "previous `this.pixelRatio`."